code/modules/rykka_puzzle_wips/reactor_defense.dm

```diff
@@ -0,0 +1,795 @@
+			warmup()
+		
+		if(ENGAGED) // All of the wave stuff is handled by procs. Go down to PROCS section to modify this behavior.
+			if(!wave_complete)
+				start_wave()
+				return // We return so that the second if isn't checked before wave_complete is set - if this is removed, wave_complete will be null/0 when the second if is run, returning true and running end_wave()
+			if(world.time >= wave_complete)
+				end_wave()
+				return
+			defense_mode()
+			
+		if(VERIFYING) // All of the verification stuff is handled by procs. Go down to PROCS section to modify this behavior.
+			if(!verification_timeout)
+				start_verification()
+				return // We return so that the second if isn't checked immediately before verification_timeout is set.
+			if(world.time >= verification_timeout && !(verification_successful)) // Did we run out of time?
+				rollback_wave()
+				return
+			else if(verification_successful)
+				complete_verification()
+				return
+			verifying()
+		
+		if(FINISHED)
+			STOP_PROCESSING(SSobj, src)
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/*													End Object Definition																		*/
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/*	//		=====							Interaction Code Begins															=====		//
+ *	//		=====			Handle Examine, attempting to interact with object, other "generic" code here.					=====		//
+ *////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/obj/structure/damaged_reactor/attack_hand(var/mob/living/user) // Is a LIVING mob clicking on us? Sorry ghosts. :p
+	if(!isliving(user)) // You're not alive. Stop. Bad.
+		return // No clicky.
+	switch(user.a_intent)
+		if(I_HELP)
+			ui_interact(user) // Open NanoUI
+		if(I_HURT)
+			var/mob/living/carbon/human/H = usr
+			if(H.species.can_shred(H)) // This is a hard vore proc. Whyyyyy???
+				attack_generic(H,10) // Handle doing damage here too, since it counts as an interaction.
+				return
+			// else if(H.species.) TODO: Figure out how to handle SHARP claws/etc for a very little bit of damage.
+			else // You can't damage it without a tool!
+				playsound(src, 'sound/effects/grillehit.ogg', 50, 1)
+				user.do_attack_animation(src)
+				user.visible_message("<span class = 'danger'>\The [user] slams their hand against \the [src], not appearing to damage it!</span>",
+									"<span class = 'danger'>You don't appear to do anything to the heavy exterior of \the [src]!</span>")
+		if(I_GRAB)
+			to_chat(user, "<span class ='notice'>You can't grab a reactor! To open the menu, use help intent!</span>")
+			return
+		if(I_DISARM)

```

Comment:
:information_source: It might be better to just open the UI if you don't plan to do anything else beyond print a message telling the player they're doing it wrong.

Reply:
This has also been corrected, I left them commented out so subtypes can see how the switch works/add interactions later if they want.